sklego/preprocessing.py

```diff
@@ -53,6 +54,48 @@ def _check_X_for_type(X):
             raise TypeError("Provided variable X is not of type pandas.DataFrame")
 
 
+class DateTimeFeatures(TransformerMixin, BaseEstimator):
+    """
+    Add common time-series features to data frame in your scikit-learn pipeline.
+    All time features supported by pandas.dt can be added (e.g. weekday, weekofyear, year,
+    quarter, month, day, hour and minute. By default it will generate all the time features. They
+    can be individually excluded.
+
+    :param date_col: name of the date column (default= 'date')
+    :param date_features: tuple of dt features
+
+    """
+
+    def __init__(self, date_col="date", date_features=("weekday", "weekofyear", "year", "quarter",
+                                                       "month", "day", "hour", "minute")):
+
+        self.date_col = date_col
+        self.date_features = date_features
+
+    def fit(self, X, y):
+        _ = X.get(self.date_col,
+                  f"Date column {self.date_col} doesn't exist. Please provide another one.")
+        if isinstance(_, str):
+            print(_)
+
+        bad_date_features = [attr for attr in self.date_features if not hasattr(pd.Series.dt, attr)]
+        if len(bad_date_features) >= 1:
+            raise ValueError(f"Date feature "
+                             f"{bad_date_features} "
+                             f"doesn't exist")
+        # TODO: check that column is date time, below doesn't work
```

Comment:
lets make sure this TODO is done before we merge though. 

maybe something like

```
pd.np.dtype(df[date_col]) == pd.np.dtype("<M8[ns]")
```